README.md
RNTrack: A Fidonet Message tracker
==================================

Disclaimer
----------

I don't support RNTrack any more. This repository is here just to preserve history.

What is this program for
------------------------

It is a so-called netmail tracker. Its purpose is to process netmail on
netmail hubs, though it can be helpful for an ordinary node. It parses PKTs
or message bases, compares messages with patterns, and does a specified
action if a message matches the pattern. With some limitations, you may also
use RNTrack for echomail.

Features
--------

* Support for popular message bases: OPUS (*.msg), JAM, Squish.
* Support for PKT 2+.
* Embedded Perl (if scripting is enabled at compile time).
* External templates.
* Portability.

Documentation
-------------

The `doc` and `samples` subdirectories contain documentation. You may also
subscribe to RU.FTRACK Fidonet echo area (in Russian).

Bug reporting
-------------

Read BUG-REPORTING text file for details.

License
-------

RNTrack is free software. It is licensed using GNU General Public License
version 2. See COPYING for the text of the license or visit
https://www.gnu.org if this file is missing.

Authors
-------

The current maintaner of RNTrack is Stas Degteff. See AUTHORS for the list of all
developers and more information.


ChangeLog
     RNTrack's changes log
     $Id$

! - Bug fixed
# - Added feature or changes

2.3.0 2023-02-09 (tag "v2.3.0")
-------------------------------
# add Msg.txt to .gitignore
! do not copy the NUL byte to password.
# add a possibility to change build target in "build_rpm".
# update "build_rpm" for work outside of "rntrack" directory.
# use -Og instead of -O2 when DEBUG=1.
# replace sourceforge with github in rntrack.spec.
# add make, gcc-c++ to BuildRequires.

2.2.0 2022-01-08 (tag "v2.2.0")
-------------------------------
! restrict the MSGID length by eight hex digits.

2.1.16 2021-01-08 (tag "v2.1.16")
---------------------------------
# update INSTALL file taking into account the migration from svn.
# add "build_rpm" for building a rntrack.rpm;
  add .gitignore to ignore "result/" directory created by "build_rpm"
# changes in rntrack.spec:
  add "BuildRequires: perl(ExtUtils::Embed)";
  use %make_build and %make_install; remove unnecessary code;
  use DEBUG=1 ENABLE_SCRIPTS=1 options.
# define ROOTDIR and use it in other variables;
  remove "result" subdirectory while running distclean.
  The subdirectory is created by `build_rpm`.
# reorganize makefiles to include "smapi/makefile.lnx"
  instead of using recursive make. Now calling the
  makefile again does not result in a rebuild. Parallel
  build is now possible.
# add DEBUG option to Makefile

2.1.15 2021-01-03 (tag "v2.1.15")
---------------------------------
# set version to 2.1.15 (tag "v2.1.15")
# change BUG-REPORTING and README.md after migration from svn
# rename README to README.md
# migrate the project from svn to git

2.1.14 2021-06-06 (Subversion rev.339)
--------------------------------------
! Update gentoo ebuilds: fix EAPI 7 support
! Update gentoo ebuilds: add forgotten ebuild for the version 2.1.10
# Update gentoo ebuilds: switch to EAPI=7, add a ebuild for the version 2.1.10

2.1.10 2020-04-27 (Subversion rev.335, tag "version-2.1.10")
------------------------------------------------------------
# add USE_MUSL, ARCH and STATIC build options for Linux
# use %HOME% environment variable in configuration sample
! get message name before using it in an error message
# bring the code to a standard form using uncrustify
  (https://github.com/uncrustify/uncrustify)
  with devel/uncrustify.cfg as configuration file

2.1.2 2020-04-22 (Subversion rev.326, tag "version-2.1.2")
----------------------------------------------------------
# update the manual according to changes made in v.2.1.0

2.1.0 2020-04-22 (Subversion rev.323, tag "version-2.1.0")
----------------------------------------------------------
# any action changing a message with Locked flag set is ignored
  and an error message is written to log
# do not pack messages when at least one of Locked, Sent,
  Archive/Sent attributes is set
# reset Received, Sent, KillSent, Local, Hold message attributes
  when packing the message to a packet
! fix date when a month name in the message header is not found
# add USE_CLANG=1 for using Clang compiler in linux

2.0.30 2020-02-14 (Subversion rev.295, tag "version-2.0.30")
------------------------------------------------------------
! fix FreeBSD build. Patch by Sergey Anohin 2:5034/10
# replace tag "version-2.0.26" by current tag

2.0.26 2020-02-12 (Subversion rev.291, tag "version-2.0.26")
------------------------------------------------------------
! remove lines from node1042 configuration sample trying
  to delete flags from a .pkt
! fix linux build with Perl
# small translation edits in English manual
! fix FreeBSD build. Patch by Sergey Anohin 2:5034/10

2.0.3 2020-01-31 (Subversion rev.260, tag "version-2.0.3")
----------------------------------------------------------
# add RNtrack configuration files from 2:5020/1042

2.0.2 2020-01-30 (Subversion rev.257)
------------------------------------
# upgrade smapi library from v.2.4.1 to v.2.5.1
! fix Visual Studio 6 build
# add Windows x64 build
# add Visual Studio 2019 solution
# update Russian manual
# make a new translation of the amended manual from Russian into English
# add manual in the form of man-pages and plain text in English
# change default path of configuration file: "~/fido/etc/rntrack.conf" under
  a UNIX-like OS, "%USERPROFILE%\fido\rntrack.cfg" under Windows or
  "rntrack.cfg" under another OS
# change the log date format to ISO 8601

1.41 2019-09-29 (Subversion rev.191)
-----------------------------------
# add a log message at LogLevel=2 with full path of the created pkt.

1.40 2019-09-29 (Subversion rev.190)
-----------------------------------
! test simple boolean value before testing a function value.
! fix compatibility with new portage version, add 1.39 version for gentoo.

1.39 2019-06-14 (Subversion rev.188)
-----------------------------------
# change fonts in Manual_ru.odt.
! fix small issues in README, INSTALL and BUG-REPORTING.
! Create solution for a build using MS Visual C++ 2010
  and fix sources for that.

1.38 2019-06-08 (Subversion rev.184)
-----------------------------------
! Fix variable initialization.
! Fix defined and/or set but unused variables.
! Fix conversion from string constant to ‘char *’.
! Fix similar issue breaking strict-aliasing rules in sqbase.cpp.
! Fix dereferencing type-punned pointer breaking strict-aliasing rules in api_sdm.c.

1.37 2019-06-07 (Subversion rev.179)
-----------------------------------
! Fix build under FreeBSD 12.0. Patch by andrew clarke 3:633/267.
! Fix build under linux (Ubuntu server 18.04 LTS). Patch by mark lewis 1:3634/12.

1.36 2018-08-25 (Subversion rev.174)
-----------------------------------
! Fix a bug in not deleting a flag after ScanDir of an empty base.
  Bug report by Alex Barinov 2:5020/715

1.35 2018-08-25 (Subversion rev.172)
-----------------------------------
! Fix a bug in setting message attributes.
  Bug report by Alex Barinov 2:5020/715

1.34 2018-08-25 (Subversion rev.171)
-----------------------------------
! Gentoo linux: fix repository path in layman config

hpp/constant.hpp
/*
 *  RNtrack - FTN message tracker/router
 *
 *  constants.hpp - Main constants definitions
 *
 *  Copyright (c) 2003-2005 Alex Soukhotine, 2:5030/1157
 *
 *  This program is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 2 of the License, or
 *  (at your option) any later version.
 *
 *  $Id$
 */

#ifndef _CONST_HPP_
#define _CONST_HPP_

#ifdef HAVE_CONFIG_H
    #include "aconfig.h"
#endif

#if !defined(__OS2__) && defined(__EMX__)
    #define __OS2__
#endif

#if defined(__NT__) || defined(__MINGW32__) || defined(_WINDOWS)
    #define OsType         "/Win"
    #define PATHDELIMS     "\\"
    #define PATHDELIMC     '\\'
#elif defined(__OS2__) || defined(__os2__)
    #define OsType         "/OS2"
    #define PATHDELIMS     "\\"
    #define PATHDELIMC     '\\'
#elif defined(__DJGPP__) || defined(__DOS4G__)
    #define OsType         "/386"
    #define PATHDELIMS     "\\"
    #define PATHDELIMC     '\\'
#elif defined (__unix__)
    #if defined (__linux__)
        #define OsType "/Lnx"
    #elif defined (__FreeBSD__)
        #define OsType "/FBSD"
    #elif defined (__OpenBSD__)
        #define OsType "/OBSD"
    #elif defined (__NetBSD__)
        #define OsType "/NBSD"
    #elif defined (__sun__)
        #define OsType "/SUN"
    #elif defined (_AIX)
        #define OsType "/AIX"
    #elif defined (__OSX__)
        #define OsType "/OSX"
    #elif defined (__PLATFORM_TYPE__)
        #define OsType "/"__PLATFORM_TYPE__
    #else
        #define OsType "/Unix"
    #endif
    #define PATHDELIMS     "/"
    #define PATHDELIMC     '/'
#endif

#ifndef PACKAGE_NAME
    #define PACKAGE_NAME "RNTrack"
#endif
#ifdef VERSION
    #undef VERSION
#endif
#define VERSION               "2.3.0"
#ifdef __PERL_VERSION__
    #define Perl              "/Perl"
#else
    #define Perl
#endif
#define ProgVersion           VERSION OsType Perl
#define ProductLO             0xff
#define ProductHI             0x16
#define Copyright \
    "\n(C) 2003-2006 Alex Soukhotine (2:5030/1157)\n    2007-2011 Stas Degteff (2:5080/102)\n    2009-2020 Michael Dukelsky (2:5020/1042)\n"

#ifndef DefaultConfig
    #if defined(__unix__)
        #define DefaultConfig "~/fido/etc/rntrack.conf"
    #elif defined(__WIN32__)
        #define DefaultConfig "%USERPROFILE%\\fido\\rntrack.cfg"
    #else
        #define DefaultConfig "rntrack.cfg"
    #endif
#endif

#ifdef __unix__
    #define MsgExtension ".msg"
    #define PktExtension ".pkt"
    #define BsyExtension ".bsy"
    #define PntExtension ".pnt"
#else
    #define MsgExtension ".MSG"
    #define PktExtension ".PKT"
    #define BsyExtension ".BSY"
    #define PntExtension ".PNT"
#endif

#define MAXCFGLINE       4096
#define DELIMETERS       " \t"
#define STRDELIMETERS    "\t\""
#define MAXPOINTS        1024

#define LOGE             100
#define LOGW             101
#define LOGI             102
#define LOGN             103
#define LOGD             105

#define BUFF_SIZE        1024
#define SMALL_BUFF_SIZE  512

#endif
